### DISCUSSION TITLE
Trying to build a menu bar

### DISCUSSION THREAD

**Discussion by clarknova** [floating-ui/floating-ui#2805] [Q&A] [answered]
I am trying to build a menubar.
I adapted this example [link](https://codesandbox.io/p/sandbox/admiring-lamport-5wt3yg?file=%2Fsrc%2FDropdownMenu.tsx) in combination with the composite component. I am basically using the render prop of composite to render each item as dropdown. Is emit "openmenu" events onNavigate to close the menu i am navigating away from. This works great with keyboard navigation but has some issues. For example: When i have one menu open and i want to navigate to the next one with keyboad or mouse-hover the element i navigate to should open automatically (basically like a native menubar). I tried to communicate the "something is open" state with events to the menubar and activate hover and focus interactions for the  root trigger but this has several issues. For once i am struggling to figure out if any element below a certain nodeId is open. It's easy to compute but i cant subscribe to it because it is a ref?
Is this generally speaking the right approach or completely misguided?

**Comment by atomiks** [comment, marked as answer]
The [`Menubar`](https://github.com/floating-ui/floating-ui/blob/master/packages/react/test/visual/components/Menubar.tsx) example locally here works with keyboard management fine, but it doesn't implement the "open on hover" behavior if one of the items is already open.

I'd have the Menubar component hold an `openOnHover` state and wrap a context provider around its rendered `Composite` and items. Inside the composite item component you're rendering, read that context and set `useHover` hook to `enabled: openOnHover` from the context. Setting it back to `false` probably occurs upon dismiss of one of the composite items, so you can pass the setter down as well.

**Reply by clarknova** [reply]
This is extremely helpful! I did not know about these examples. Maybe this folder should be mentioned on the documentation page under React Examples.
